odxtools/flashdata.py

```diff
@@ -30,31 +30,33 @@ def data_str(self) -> str:
                                   f"by the {type(self).__name__} class")
 
     @property
-    def dataset(self) -> IntelHexDataSet | MotorolaSDataSet | bytearray | None:
+    def dataset(self) -> BinFile | bytearray | None:
```

Comment:
You can create a BinFile from bytearray to have consistent return value

Reply:
that's the purpose of the `.blob` property. `.dataset` allows you to access additional information provided by the underlying dataset, e.g. the execution start address...